<!doctype html> <html lang=en > <meta charset=UTF-8 > <meta name=viewport  content="width=device-width, initial-scale=1"> <link rel=stylesheet  href="/css/franklin.css"> <link rel=stylesheet  href="/css/hypertext.css"> <link rel=icon  href="/assets/favicon.png"> <title>schu - About</title> <header> <h1 style="color:#283747">Shane Chu</h1> <nav> <a href="/" class=current >About</a> | <a href="/research">Research</a> | <a href="/blog">Research notes</a> | <a href="/tags/">Tags</a> | <a href="/Shane_CV.pdf">CV</a> <hr/> </nav> </header> <div class=franklin-content ><div class=img-me ><img src="./me.jpg" alt=Proguard  /></div> <p>I am a post-doctoral research scientist at Columbia University, collaborating with Chaolin Zhang other wonderful members in the <a href="https://zhanglab.c2b2.columbia.edu/">Zhang Lab</a>. Before this, I completed my Ph.D. in computer science at Washington University in St. Louis under the guidance of Gary Stormo. My research interest is developing computational methods for problems related to protein-DNA/RNA interactions.</p> <p>A central theme in my research is to consider what representation is suited for DNA sequences that give the most helpful information. Traditionally, there are k-mers, position-weight-matrices, parametric statistical models like HMMs, and, more recently, deep neural networks. I am particularly interested in sparse representations, a rich framework that allows us to build techniques to answer challenging inferential questions in regulatory genomics.</p> <p>For example, in regulatory genomics, we want to infer the following:</p> <ul> <li><p>The DNA sequences&#39; regulatory elements, e.g., the motifs.</p> <li><p>The key regulatory elements in the DNA sequences give rise to the observed phenomenon.</p> <li><p>The counterfactuals, e.g., a minimal change to a DNA sequence that turns off the observed effects.</p> </ul> <p>We can obtain a sparse representation of DNA sequences using principled deep learning techniques, e.g., deep unfolding. We use <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btad378/7192989?utm_source&#61;advanceaccess&amp;utm_campaign&#61;bioinformatics&amp;utm_medium&#61;email">sparse representations to reveal many more hidden motifs not shown on the JASPAR database</a>. Our result shows that sparse representation is a scalable and interpretable approach to biological sequence problems.</p> <p>My name in Mandarin is Kuei-Hsien Chu &#40;朱桂賢&#41;. Hsien pronounced like Shane, and that&#39;s how I got Shane. I grew up in Taiwan and have been living in the US since 2008.</p> <div class=page-foot > Contact me by <a href="mailto:skchu@wustl.edu">E-mail</a> | <a href="https://github.com/kchu25">Github</a> | <a href="https://www.linkedin.com/in/kchu1/">Linkedin</a> <br> This work is licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a>. Last modified: May 05, 2024. <br> Website built with <a href="https://github.com/tlienart/Franklin.jl">Franklin.jl</a> and the <a href="https://julialang.org">Julia language</a>. </div> </div>