<!doctype html> <html lang=en > <meta charset=UTF-8 > <meta name=viewport  content="width=device-width, initial-scale=1"> <link rel=stylesheet  href="/libs/katex/katex.min.css"> <link rel=stylesheet  href="/css/franklin.css"> <link rel=stylesheet  href="/css/hypertext.css"> <link rel=icon  href="/assets/favicon.png"> <title>C2H2 zinc finger as a genome editing tool</title> <header> <h1 style="color:#283747">Shane Chu</h1> <nav> <a href="/" class=current >About</a> | <a href="/research">Research</a> | <a href="/blog">Research notes</a> | <a href="/tags/">Tags</a> | <a href="/Shane_CV.pdf">CV</a> <hr/> </nav> </header> <div class=franklin-content ><h3 id=c2h2_zinc_finger_as_a_genome_editing_tool ><a href="#c2h2_zinc_finger_as_a_genome_editing_tool" class=header-anchor >C2H2 zinc finger as a genome editing tool</a></h3> <div class=img-small ><img src="/blog/pics/zf.png" alt=Zinc-Finger  /></div> <blockquote> <p><a href="https://en.wikipedia.org/wiki/Zinc_finger">C2H2 zinc finger protein</a>.</p> </blockquote> <p>At this time of writing, using a Zinc finger &#40;C2H2-ZF&#41; to edit DNA sequences sounds ridiculous.</p> <p>There are much easier tools to edit DNA sequences, e.g., CRISPR Cas 9.</p> <p>Like its name, C2H2-ZF proteins use their &quot;fingers&quot; to interact with the DNA. We now have pretty standard technologies to make the &quot;fingers&quot; of C2H2-ZF, i.e., the fingers are now our lego bricks. The bottleneck is that we don&#39;t have a clear picture of how the composition of the &quot;fingers&quot; of a C2H2-ZF affects its DNA binding sites.</p> <p>Exhaustively checking the different compositions of C2H2-ZF and their binding sites is not an option since protein engineering is expensive.</p> <h4 id=recognition_code ><a href="#recognition_code" class=header-anchor >Recognition code</a></h4> <p>From a computer science perspective, the above senario is a task of characterizing the following map</p> <span class=katex-display ><span class=katex ><span class=katex-mathml ><math xmlns="http://www.w3.org/1998/Math/MathML" display=block ><semantics><mrow><mi mathvariant=bold-italic >f</mi><mo>:</mo><mrow><mtext mathvariant=sans-serif >amino</mtext><mtext> </mtext><mtext mathvariant=sans-serif >acid</mtext><mtext> </mtext><mtext mathvariant=sans-serif >composition</mtext><mtext> </mtext><mtext mathvariant=sans-serif >of</mtext><mtext> </mtext><mtext mathvariant=sans-serif >a</mtext><mtext> </mtext><mtext mathvariant=sans-serif >C2H2-ZF</mtext></mrow><mo>→</mo><mrow><mtext mathvariant=sans-serif >binding</mtext><mtext> </mtext><mtext mathvariant=sans-serif >sites</mtext></mrow></mrow><annotation encoding="application/x-tex">\bm f : \textsf{amino acid composition of a C2H2-ZF} \rightarrow \textsf{binding sites}</annotation></semantics></math></span><span class=katex-html  aria-hidden=true ><span class=base ><span class=strut  style="height:0.8889em;vertical-align:-0.1944em;"></span><span class=mord ><span class=mord ><span class="mord boldsymbol" style="margin-right:0.11042em;">f</span></span></span><span class=mspace  style="margin-right:0.2778em;"></span><span class=mrel >:</span><span class=mspace  style="margin-right:0.2778em;"></span></span><span class=base ><span class=strut  style="height:0.8889em;vertical-align:-0.1944em;"></span><span class="mord text"><span class="mord textsf">amino acid composition of a C2H2-ZF</span></span><span class=mspace  style="margin-right:0.2778em;"></span><span class=mrel >→</span><span class=mspace  style="margin-right:0.2778em;"></span></span><span class=base ><span class=strut  style="height:0.8889em;vertical-align:-0.1944em;"></span><span class="mord text"><span class="mord textsf">binding sites</span></span></span></span></span></span> <p>where the map <span class=katex ><span class=katex-mathml ><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi mathvariant=bold-italic >f</mi></mrow><annotation encoding="application/x-tex">\bm f</annotation></semantics></math></span><span class=katex-html  aria-hidden=true ><span class=base ><span class=strut  style="height:0.8889em;vertical-align:-0.1944em;"></span><span class=mord ><span class=mord ><span class="mord boldsymbol" style="margin-right:0.11042em;">f</span></span></span></span></span></span> is called the <em>recognition code</em> in biology. The recognition code is very much a machine-learning problem. Several established tools exist, such as the <a href="http://zf.princeton.edu/logoMain.php">Interactive PWM predictor</a> and <a href="http://rcade.ccbr.utoronto.ca/Help.html">RCADE</a>, which seek to characterize the recognition code. Both tools characterize the binding sites as <a href="https://en.wikipedia.org/wiki/Position_weight_matrix">position weight matrices &#40;PWM&#41;</a>.</p> <p>Whether to characterize the binding sites in the recognition code problem as a PWM is an open research question. PWM is a compact representation of the binding sites and a <a href="../dist_rep/">local representation</a>. It offers easy interpretations but at the same time, making simplifying assumptions –– e.g., an independent contribution from each position to the binding specificity. Some argue that we need a more expressive representation to reveal more biological insights.</p> <h4 id=whats_advantageous_about_c2h2_zinc_fingers_for_genome_editing ><a href="#whats_advantageous_about_c2h2_zinc_fingers_for_genome_editing" class=header-anchor >What&#39;s advantageous about C2H2 zinc fingers for genome editing?</a></h4> <p>C2H2 zinc fingers are smaller than other genome editing tools, giving it a more straightforward delivery process into the cell via viral vectors.</p> <hr /> <p>Update &#40;4/7/2023&#41;:</p> <p>Recently, an article was published about using ZF as a recognition code. I haven&#39;t looked into it in detail; if you are interested, see <a href="https://www.nature.com/articles/s41587-022-01624-4">A universal deep-learning model for zinc finger design enables transcription factor reprogramming</a>.</p> <h4 id=references ><a href="#references" class=header-anchor >References</a></h4> <ul><li id="#gupta2014expanding">Rajat M and Musunuru Gupta, <span style="font-style:italic;">Expanding the genetic editing tool kit: ZFNs , TALENs , and CRISPR-Cas9</span>, 2014.<li id="#najafabadi2015identification">Hamed S and Albu Najafabadi, <span style="font-style:italic;">Identification of C2H2-ZF binding preferences from ChIP-seq data using RCADE</span>, 2015.<li id="#persikov2009">Anton V and Osada Persikov, <span style="font-style:italic;">Predicting DNA recognition by Cys2His2 zinc finger proteins</span>, 2009.<li id="#ichikawa2023universal">David M and Abdin Ichikawa, <span style="font-style:italic;">A universal deep-learning model for zinc finger design enables transcription factor reprogramming</span>, 2023.</ul> <div class=page-foot > Contact me by <a href="mailto:skchu@wustl.edu">E-mail</a> | <a href="https://github.com/kchu25">Github</a> | <a href="https://www.linkedin.com/in/kchu1/">Linkedin</a> <br> This work is licensed under <a href="http://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a>. Last modified: September 07, 2023. <br> Website built with <a href="https://github.com/tlienart/Franklin.jl">Franklin.jl</a> and the <a href="https://julialang.org">Julia language</a>. </div> </div>